inc/class-clean-emails.php

```diff
@@ -86,6 +87,11 @@ public function comment_notification_text( $message, $comment_id ) {
 	public function comment_moderation_text( $message, $comment_id ) {
 		$this->setup_data( $comment_id );
 
+		if ( empty( $this->comment->comment_type ) ) {
+			$this->comment->comment_type = __( 'comment', 'yoast-comment-hacks' );
+		}
+
+		/* translators: %1$s is replaced with the comment type, %2$s is replace with the post title */
 		$this->message = sprintf( __( 'A new %1$s on the post "%2$s" is waiting for your approval:', 'yoast-comment-hacks' ), $this->comment->comment_type, '<a href="' . get_permalink( $this->comment->comment_post_ID ) . '">' . esc_html( $this->post->post_title ) . '</a>' ) . '<br /><br />';
```

Comment:
As a best practice, WordPress avoids to use post/comment types as placeholder values since this would make translations difficult in some cases. For example, `%1$s` can be replaced by comment/pingback/trackback and in some languages they could be words with a different genre, making impossible to correctly translate "new". I'd suggest a `switch` statement and split this in 3 different strings:
`A new trackback ...`
`A new pingback ...`
`A new comment ...`
Ideally, the same should be done for `comment_notification_text()`.